tests/test_AggregateWellDataLGR.cpp

```diff
@@ -956,13 +956,10 @@ BOOST_AUTO_TEST_CASE (Declared_Well_Data2LGRWells)
     const auto rptStep = std::size_t{1};
 
     auto countWells = [&simCase](const std::string& lgr_tag) -> int {
-        int num_filtered_wells = 0;
-        for (const auto& well : simCase.sched.getWells(rptStep)) {
-            if (well.get_lgr_well_tag().value_or("") == lgr_tag) {
-                ++num_filtered_wells;
-            }
-        }
-        return num_filtered_wells;
+        const auto wells = simCase.sched.getWells(rptStep);
```

Comment:
It's a unit test so performance doesn't really matter all that much, but `Schedule::getWells(step)` generates a `vector<Well>` that contains copies of all the pertinent `Well` objects stored in the `Schedule`.  If it were me, I'd probably use `Schedule::wellNames(step)` instead.  For instance something along the lines of
```C++
const auto wnames = simCase.sched.wellNames(rptStep);
return std::count_if(wnames.begin(), wnames.end(),
    [&sched = simCase.sched[rptStep], &lgr_tag](const std::string& wname)
{
    return sched.wells(wname).get_lgr_well_tag().value_or("") == lgr_tag;
});
```